### DISCUSSION TITLE
Mastodon-based comment system?

### DISCUSSION THREAD

**Discussion by Andre601** [squidfunk/mkdocs-material#5728] [Q&A] [answered]
I would like to move away from GitHub and GitHub Pages with my blog and instead move to Codeberg Pages.

To achieve a complete migration would I like to also move the comment system away from giscuss to Mastodon.
I do like Giscus, but I don't want people to require a GitHub Account just for commenting while they may have a Mastodon account instead...

Tho, there is a kinda obvious question I would need answers for first: How would I implement this?

I've found [a blog post](https://danielpecos.com/2022/12/25/mastodon-as-comment-system-for-your-static-blog/) talking about a Mastodon-based comment system. The issue I see with that is, that it's about Hugo, so not idealy the same as MkDocs or Material for MkDocs.

I would (probably) also need to create some setup to automatically create a necessary toot on my Mastodon instance for the comment system to work...

In either case am I not experienced enough to tackle this, so I would apreciate if anyone could help me here.
I really want to make this move, but this part is making me struggle a bit...

**Comment by Andre601** [comment]
My first attempt using [this code snippet here](https://carlschwan.eu/2020/12/29/adding-comments-to-your-static-blog-with-mastodon/) is without any success... And I can't for the life of me understand the stacktrace python throws at me... I think it's related to my `{% if ... %}` line to make sure only posts with an associated mastodon post work...

Full comments.html: https://paste.helpch.at/ojojujojeg.html

Exception: https://paste.helpch.at/gecudecahi.py

**Comment by Andre601** [comment, marked as answer]
Managed to get a working solution using the comment example from the blog post I mentioned in my initial comment: https://danielpecos.com/2022/12/25/mastodon-as-comment-system-for-your-static-blog/

All the files including a simple setup guide: https://paste.gg/p/Andre601/4d1323b59c35421e90bcdc79f6ed8ff2

## How it looks

No comments yet:
![grafik](https://github.com/squidfunk/mkdocs-material/assets/11576465/e14a66f8-714f-4aca-987e-e074f0b0e5a3)

No ID found:
![grafik](https://github.com/squidfunk/mkdocs-material/assets/11576465/7521a64f-5899-49e0-8797-6e7ee262e738)

Actual comment:
![grafik](https://github.com/squidfunk/mkdocs-material/assets/11576465/31f7b5ad-8adf-4866-8ee1-aa85a4757809)

**Reply by squidfunk** [reply]
Thanks for sharing!

**Reply by Andre601** [reply]
No problem... Tho I wonder if there would be room for further optimization. JS isn't my strong suit, so I can't tell if that's the best it can be or if there's other ways that are more optimized, both for the site and the Mastodon instance (As in reducing possible API spam on their end).

**Reply by Andre601** [reply]
Also, random question, but would you be willing to have a section added in the Comment system setup page?
The section would only mention requirements and then link to a paste or (probs better) a github repo with further instructions and the files...

Although, I kinda want to first improve the CSS a bit to use the actual colours from Material for MkDocs to not have duplicate values... But for some odd reason is it not accepting just the var name and seems to require from me to setup a theme-specific entry (`@media screen [data-md-color-scheme="slate"]`) or smth... It's really weird...

Maybe there's a way to import the CSS from Material for MkDocs in some way? Tho given it's minimized and re-generated because of that, this could be problematic...

**Reply by squidfunk** [reply]
> Also, random question, but would you be willing to have a section added in the Comment system setup page?

Currently not, as we consider this heavily being in prototype stage – nothing we'd like to advertise and provide support for right now. This might change in the future when there's more adoption and maybe even a simpler way to integrate it.

**Reply by Andre601** [reply]
I'm now wondering... Would it be better to put parts of the JS code into a JS file and add it as extra javascript in the mkdocs.yml?

Not sure what optimizations I could do here to not inject/copy-paste the entire thing into every page using it... Headache for sure.
